### ISSUE TITLE
Fail if completed season doesn't line up with results.

### ISSUE THREAD

**Issue by AllRoCol** [Sonarr/Sonarr#4563]
**Is your feature request related to a problem? Please describe.**

The issue is that many shows, far more than thexem has, don't line up at all. A lot of it comes from streaming networks altering shows once they reach them or the dreaded double episode handling. Having put together a collection once, of over 5000 shows, all manually searched, I have caught many over the years (Too many to count). Easily a triple digit amount. Usually it goes something like 2 part mini-series becomes 4 parts or old web show has episodes combined. Obviously there are big ones like Money Heist and Close Enough, but there are so may small ones that one will get lost in them. Even something known like Terra Nova causes problems due to it being released with too many episodes, forcing a combining of the episodes manually. The system still tries to grab it, which then makes matters way worse. 

This is a growing problem caused by incompetence in handling databases (Especially with double episodes), combined with the streaming networks deciding to do whatever they want. Neither one is every going to stop, the former problem goes back to the beginning of TV show databases and the latter problem is growing by the day. You may not notice it as much if you don't venture into the obscure, but it is there big time. For example, Amazon has many mini-series and TV movies that they have altered the episode count of. Probably more than 100 at this point.

Without getting fed the proper end state data (Like TVDB maybe pushing the finalized order in the API), the only thing you can do is hand alter the collection to match things up if you want to get the highest quality offerings. Once stuff reaches Blu-Ray or streaming, that is the sought after quality and the typical source of the problem. Not knowing if it needs to be hand altered causes problems.

**Describe the solution you'd like**

I would like to see any completed season that doesn't line up, episode count-wise, to fail to grab it and possibly issue a warning. Seeing a warning in the manual search results as well would be useful, since they exist for many other things. If it scans 11 episodes, but needs 13 or if it scans 13 episodes, but needs 11, then it would be nice to know, rather than have it attempt to just go ahead with it. I know enough to correct it often, but I miss a lot of things. Someone who doesn't have a big background in manually seeing this stuff will probably rarely ever catch it until they go to watch it.

The system knows which titles match most of the time and can read the episode numbers most of the time, so I hope that it could possibly detect having too many or too few. I have no expectations, to place on any of your or the system, period, but that would be as far as I would think it would or could realistically go. Until proper orders can reach an API somewhere. 

Too many is more important than too few. Too few you can notice. Too many and you may just think you have everything and find out later on that you got half episodes or that you had to hand combine stuff.

I would expect it wouldn't kick in until a certain amount of time after a season is finalized. Maybe a week or a month or something. It usually isn't needed right off, but eventually there is the question of what will it be when it hits the higher quality offerings. Some stuff jumps to streaming quickly, though.

**Describe alternatives you've considered**

The only alternative is to manual search out everything, read the results and figure out what doesn't line up. At that point you have defeated the purpose of automation.

**Additional context**

Thank you for any consideration. I apologize if this has been asked before. I tried searching for it, but hitting the right keywords didn't really happen.

**Comment by markus101** [comment]
Sonarr will leave downloads that don't have the correct number of episodes imported in the queue for confirmation after the fact, if the release has more episode files then expected then it won't do anything, but the files will not be removed from disk. At this time we do not have plans to enhance this behaviour further.

**Comment by ABoringUser** [comment]
Thank you for taking the time to respond. I wish you never having to run into this problem in the future. It is a rough one.  I hate the unknown and I have seen this problem on over 100 separate occasions, through manual downloading. Now that I am using Sonarr and replacing everything in my collection, I am praying I can remember to manually search for all of those items when I see them come up.

I understand though. It isn't a pressing issue for most, as common items only have it here and there. I only started seeing regularly when I started digging a lot deeper and getting various mini-series and TV movie offerings. I didn't think it big enough that you would consider it, but I was secretly hoping.

Thanks again.
